### PR TITLE
Added the ban API now in a different way

### PR DESCRIPTION
**Pull request by pschichtel**
I added the new methods banPlayer(String name, String message), unbanPlayer(String name), banIp(String ip) and unbanIp(String ip) tp the Server class and added the new events BAN_PLAYER (called, when a player gets banned) and BAN_IP (called when an IP gets banned). The events are cancellable, so it is possible to abort a ban.

CraftBukkit pull-request: https://github.com/Bukkit/CraftBukkit/pull/266

### CODE REVIEW

File: src/main/java/org/bukkit/event/server/BanIpEvent.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    
+    /**
+     * Returns the IP address
+     * 
+     * @return the IP
+     */
+    public String getIp() {
+        return this.ip;
+    }
+
```
Comment (tonybruess) [inline src/main/java/org/bukkit/event/server/BanIpEvent.java]: Maybe an option to see who sent the ban event?

Comment (pschichtel) [inline src/main/java/org/bukkit/event/server/BanIpEvent.java]: That's not possible here, but the plugins using the API can handle that.

**Comment by pschichtel** [comment]
did anyone of the bukkit team review the pull by now?

File: src/main/java/org/bukkit/event/server/BanIpEvent.java (hunk 1 above)
Comment (tonybruess) [inline src/main/java/org/bukkit/event/server/BanIpEvent.java]: OK, sounds good.

**Comment by pschichtel** [comment]
I'll implement this for 1.6

**Comment by pschichtel** [comment]
fail, I closed the wrong pull request

**Pull request merged**